code/modules/pda/PDA.dm

```diff
@@ -165,15 +165,16 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(usr, "<span class='notice'>You cannot do this while restrained.</span>")
 
 /obj/item/pda/AltClick(mob/user)
-	..()
+	if(!HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) && ((src in user) || (isturf(loc) && Adjacent(src, user))))
```

Comment:
TRAIT_HANDS_BLOCKED needs to be applied to the check for adjacency too

Reply:
think you're reading this wrong, it's checking for the trait first and then checking if it's either on a turf and you're adjacent or it's in your contents

Reply:
guh

Reply:
This is quite unreadable yeah. Can you make this into something more readable or add comments explaining what it does? Also making a proc of it would help since it is used twice already, should also help with readability.

Reply:
Agreed, please do this @GDNgit